frontend/src/lib/components/BillingAlerts.tsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react'
+import { useValues } from 'kea'
+import { userLogic } from 'scenes/userLogic'
+import { WarningOutlined, ToolFilled } from '@ant-design/icons'
+import { Button, Card } from 'antd'
+import { billingLogic } from 'scenes/billing/billingLogic'
+import { LinkButton } from './LinkButton'
+
+export function BillingAlerts(): JSX.Element {
+    const { user } = useValues(userLogic)
+    const { alertToShow, percentage, strokeColor } = useValues(billingLogic)
+
+    if (!alertToShow) {
+        return <></>
+    }
+
+    return (
+        <>
+            <div style={{ marginTop: 32 }} />
+            {alertToShow === 'setup_billing' && (
+                <Card>
+                    <div style={{ display: 'flex' }}>
+                        <div style={{ flexGrow: 1, display: 'flex', alignItems: 'center' }}>
+                            <WarningOutlined className="text-warning" style={{ paddingRight: 8 }} />
+                            <b>Action needed!&nbsp;</b>
+                            {user?.billing?.plan?.custom_setup_billing_message ||
+                                'Please finish setting up your billing information.'}
+                        </div>
+                        <div style={{ display: 'flex', alignItems: 'center' }}>
+                            <Button type="primary" href={user?.billing?.subscription_url} icon={<ToolFilled />}>
+                                Set up now
+                            </Button>
+                        </div>
+                    </div>
+                </Card>
+            )}
+            {alertToShow === 'usage_near_limit' && (
+                <Card>
+                    <div style={{ display: 'flex' }}>
+                        <div style={{ flexGrow: 1, display: 'flex', alignItems: 'center' }}>
+                            <WarningOutlined className="text-warning" style={{ paddingRight: 16 }} />
```

Comment:
If I have to come up with a nit on a quick skim of this is that you could have the conditional rendering for the alert content only, inside those first 3 tags which are repeated across both:

```
<Card>
     <div style={{ display: 'flex' }}>
            <div style={{ flexGrow: 1, display: 'flex', alignItems: 'center' }}>
                 {/* conditional alert content render */}
            </div>
     </div>
 </Card>

Reply:
I'll keep it right now because we also differentiate on the call to action element so we'd have to go a level above, to the initial div, and we'd have to add a condition check for when `alertToShow = null`, so seems like we land on the same complexity.